src/Bicep.Core/Parser/Parser.cs

```diff
@@ -198,26 +198,26 @@ private Token NewLine()
             return Expect(TokenType.NewLine, b => b.ExpectedNewLine());
         }
 
-        public SyntaxBase Expression()
+        public SyntaxBase Expression(bool allowComplexLiterals)
```

Comment:
Had to do this to block arrays and objects in string interpolation at the parser level because the object error recovery was eating too many tokens. We never allowed complex literals in string interpolation, we're not losing anything here.